install script updates

SticksDev opened the pull request:
Changed the macOS and Linux scripts to end in .sh
Added Windows setup script. Powershell-based, tested against Windows 11 

Let me know if you have any questions

QuinceTart10 commented:
I wish you looked at my pull request, where I already addressed the issues with the Linux and macOS install scripts and even improved them 😕

This PR should've been just the Windows install script in my opinion.

ByzantineProcess commented:
Fails on Windows 11 Sandbox:
pkg-config not found

SticksDev commented:
Hey @ByzantineProcess - that was the environment I tested in. Does the script itself fail or does the compile of Webx fail? I can't seem to reproduce.

SticksDev commented:
> I wish you looked at my pull request, where I already addressed the issues with the Linux and macOS install scripts and even improved them 😕
> 
> This PR should've been just the Windows install script in my opinion.

Oh, crap. I'll undo that then. Didn't mean to step over you, I just added .sh to the end. lol

face-hh commented:
This installs the msvc toolchain by default even though only the gnu toolchain is needed. So we end up downloading 2 toolchains, which on the VM I'm testing the script on is taking a lot of time

Does `rustup-init.exe` support CLI commands for the default toolchain? I remember running it and it had an option for that.
If it does, we could modify this line of code
```ps1
Start-Process -FilePath $rustup -ArgumentList "-y" -Wait
```

ByzantineProcess commented:
> Hey @ByzantineProcess - that was the environment I tested in. Does the script itself fail or does the compile of Webx fail? I can't seem to reproduce.

It's the compilation of Webx that fails.

ByzantineProcess commented:
> This installs the msvc toolchain by default even though only the gnu toolchain is needed. So we end up downloading 2 toolchains, which on the VM I'm testing the script on is taking a lot of time
> 
> Does `rustup-init.exe` support CLI commands for the default toolchain? I remember running it and it had an option for that. If it does, we could modify this line of code
> 
> ```powershell
> Start-Process -FilePath $rustup -ArgumentList "-y" -Wait
> ```

In my testing, (unless it happened very quickly and silently), it didn't seem to install MSVC. I don't have install logs to double-check, I just don't remember Rust trying to install it, as it didn't block the install waiting for the user to install it themselves like it did on another machine I installed napture on.

neoapps-dev commented:
mine got rewritten 🗿
